etna/core/utils.py

```diff
@@ -0,0 +1,39 @@
+import inspect
+from copy import deepcopy
+from functools import wraps
+from typing import Callable
+
+
+def init_collector(init: Callable) -> Callable:
+    """Make decorator for collecting init parameters."""
+
+    @wraps(init)
+    def wrapper(*args, **kwargs):
+        self, *args = args
+        init_args = inspect.signature(self.__init__).parameters
+
+        deepcopy_args = deepcopy(args)
+        deepcopy_kwargs = deepcopy(kwargs)
+
+        self._init_params = {}
+        args_dict = dict(
+            zip([arg for arg, param in init_args.items() if param.kind == param.POSITIONAL_OR_KEYWORD], deepcopy_args)
```

Comment:
Can we meet problems with other kinds? E.g. `kwargs` parameters: [docs](https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind).

Reply:
The will be in kwargs

Reply:
Will it work with `POSITIONAL_ONLY` somehow?

Reply:
No

Reply:
I think we should write in documentation about the limitations of this approach.

Reply:
I've added comment